python/Scripts/creatematerial.py

```diff
@@ -0,0 +1,235 @@
+    with open(json_path, 'r') as f:
+        data = json.load(f)
+    return data
+
+
+def texture_type_from_name(texture_name: str) -> List[str]:
+    """
+    Get the texture matched pattern and return the materialx plug name and the plug type
+    @param texture_name: The texture filename
+    return: list(material_input_name, material_input_type)
+    """
+    settings = get_cfg()
+    texture_patterns = settings.get('patterns')
+
+    # reversed to take the color before weight e.g. specular_color before specular
+    for plug_name in reversed(list(texture_patterns.keys())):
+        regx_list = texture_patterns[plug_name]['values']
+        if not isinstance(regx_list, list):
+            continue
+
+        for regex in regx_list:
+            if re.search(regex, texture_name):
+                return [plug_name, texture_patterns[plug_name]['type']]
+
+
+def list_textures(texture_dir: MaterialX.FilePath) -> List[udimutils.UDIMFile]:
+    """
+    List all textures that matched extensions in cfg file
+    @param texture_dir: the directory where the textures exist
+    return List(udimutils.UDIMFile)
+    """
+    tex_extensions = get_cfg().get('texture_extensions')
+    all_textures = []
+    for ext in tex_extensions:
+        textures = [texture_dir / f for f in texture_dir.getFilesInDirectory(ext)]
+
+        while textures:
+            texture_file = udimutils.UDIMFile(textures[0].asString())
+            all_textures.append(texture_file)
+            for udim_file in texture_file.getUdimFiles():
+                textures.remove(udim_file)
+
+    return all_textures
+
+
+def create_mtlx_doc(texture_dir: MaterialX.FilePath,
+                    mtlx_file: MaterialX.FilePath,
+                    relative_paths: bool = True,
+                    colorspace: str = 'srgb_texture',
+                    use_tile_image: bool =False
+                    ) -> None:
+    """
+    Create a metrical document with uber shader
+    @param texture_dir: The texture path directory
+    @param mtlx_file: The output path of document
+    @param relative_paths: Will create relative texture path or not inside document
+    @param colorspace: The given colorspace
+    @param use_tile_image: use tileimage node or image node
+    """
+
+    material_type = 'standard_surface'
+
+    # Create document
+    doc = MaterialX.createDocument()
+
+    mtlx_filename = mtlx_file.getBaseName().rsplit('.', 1)[0]
+    mtlx_filename = doc.createValidChildName(mtlx_filename)
+
+    # Create node graph and material
+    graph_name = doc.createValidChildName('NG_' + mtlx_filename)
+    node_graph = doc.getNodeGraph(graph_name)
+    if not node_graph:
+        node_graph = doc.addNodeGraph(graph_name)
+
+    shader_node = doc.addNode(material_type, 'SR_' + mtlx_filename, 'surfaceshader')
+    doc.addMaterialNode('M_' + mtlx_filename, shader_node)
+
+    udim_numbers = set()
+    texture_files = list_textures(texture_dir)
+
+    if not texture_files:
```

Comment:
Minor: Might want to do this firewall check before creating any content for the doc.

Reply:
Did you mean if we have permission to write in a certain directory?